R/method-correlation.R

```diff
@@ -0,0 +1,125 @@
+#' Correlation
+#'
+#' This function returns correlation coefficients for variables in one dataset against variables in a second dataset
+#' 
+#' @param data1 first dataset. An AbundanceData object or data.table
+#' @param data2 second dataset. A SampleMetadata object (if data1 is class AbundanceData) or a data.table
+#' @param method string defining the type of correlation to run. The currently supported values are 'spearman' and 'pearson'
+#' @param verbose boolean indicating if timed logging is desired
+#' @return ComputeResult object
+#' @import veupathUtils
+#' @import data.table
+#' @useDynLib microbiomeComputations
+#' @export
+setGeneric("correlation",
+  function(data1, data2, method = c('spearman','pearson'), verbose = c(TRUE, FALSE), ...) standardGeneric("correlation"),
+  signature = c("data1","data2")
+)
+
+
+#' Correlation
+#'
+#' This function returns correlation coefficients for all columns in one data table with all columns in a second data table.
+#' 
+#' @param data1 data.table with columns as variables. All columns must be numeric. One row per sample.
+#' @param data2 data.table with columns as variables. All columns must be numeric. One row per sample. Will correlate all columns of data2 with all columns of data1.
+#' @param method string defining the type of correlation to run. The currently supported values are 'spearman' and 'pearson'
+#' @param verbose boolean indicating if timed logging is desired
+#' @return ComputeResult object
+setMethod("correlation", signature("data.table", "data.table"), function(data1, data2, method = c('spearman','pearson'), verbose = c(TRUE, FALSE)) {
```

Comment:
my plan is to add methods to this function that support compositional data. The AbundanceData would probably need an isCompositional flag and then if we try to run a non-compositional method on compositional data it could err. Or there could be a compositional subclass and it gets a method. Either way i think the way it's set up will make adding the compositional methods pretty "simple"!

Reply:
Are we simply not worried about it for Dec? I'd guess we need it for release though no? Since compositional data are critical here..